examples/xgboost/README.md

```diff
@@ -8,15 +8,89 @@ They use [XGBoost](https://github.com/dmlc/xgboost),
+
+1. uniform: all clients has the same amount of data 
+2. linear: the amount of data is linearly correlated with the client ID (1 to M)
+3. square: the amount of data is correlated with the client ID in a squared fashion (1^2 to M^2)
+4. exponential: the amount of data is correlated with the client ID in an exponential fashion (exp(1) to exp(M))
+
+The choice of data split depends on dataset and the number of participants. For a large dataset as HIGGS, if the number of clients is small (e.g. 5), each client will still have sufficient data to train on with uniform split, and hence exponential would be used to observe the performance drop caused by non-uniform data split. If the number of clients is large (e.g. 20), exponential split will be too aggressive, and linear/square should be used.
+
+Data splits used in the following experiment can be generated with
+```
+bash data_split_gen.sh
+```
+To be specific, this script calls the python script `./utils/prepare_data_split.py`. The arguments are:
+- site_num: total number of sites
+- site_name: site name prefix
+- size_total: total number of instances, for HIGGS dataset it is 11 million
+- size_valid: validation size, for the experiments here, it is 1 million, indicating the first 1 million instances will be used as standalone validation set. 
+- split_method: how to split the dataset, can be uniform, linear, square, and exponential
+- out_path: output path for the data split json file 
+This will generate data splits for two client sizes: 5 and 20, and 3 split conditions: uniform, square, and exponential. Users can further customize it for more experiments.
+> **_NOTE:_** The generated train config files will be stored in the folder `./data_splits`, and will be used by job_configs by specifying the path within `config_fed_client.json` 
+
+
+### Prepare job configs under various training schemes
+We then prepare the job configs for NVFlare jobs corresponding to various settings by running
+```
+bash job_config_gen.sh
+```
+To be specific, this script calls the python script `./utils/prepare_job_config.py`. It modifies settings from a base config `./job_configs/higgs_base`, and copies the correct data split file generated in the data preparation step.
+
+Here, we generated in total 10 different configs: five for each of the 5/20-client settings:
+- cyclic training with uniform data split 
+- cyclic training with non-uniform data split 
+- bagging training with uniform data split and uniform shrinkage 
+- bagging training with non-uniform data split and uniform shrinkage 
+- bagging training with non-uniform data split and scaled shrinkage
+
+By default, cpu based training is used.  For gpu based training, edit `job_conf_gen.sh` to change `TREE_METHOD="hist"` to `TREE_METHOD="gpu_hist"`.
 
+> **_NOTE:_** Cyclic training always use uniform shrinkage. The generated job config files will be stored in the folder `./job_configs`
```

Comment:
I guess now it will have a new path configuration to distinguish from histogram-based?

Reply:
They both shared the same job_gen_config.sh